pkg/prometheus/agent/operator.go

```diff
@@ -82,14 +83,64 @@ type Operator struct {
 	endpointSliceSupported bool
 }
 
+func Runnable(ctx context.Context, conf operator.Config, logger log.Logger) (bool, error) {
+	cfg, err := createK8sClusterConfig(conf)
+	if err != nil {
+		return false, errors.Wrap(err, "instantiating cluster config failed")
+	}
+
+	kclient, err := createK8sClient(cfg)
+	if err != nil {
+		return false, errors.Wrap(err, "instantiating kubernetes client failed")
+	}
+
+	crdInstalled, err := k8sutil.IsAPIGroupVersionResourceSupported(kclient.Discovery(), monitoringv1alpha1.SchemeGroupVersion.String(), monitoringv1alpha1.PrometheusAgentName)
+	if err != nil {
+		level.Warn(logger).Log("msg", "failed to check if the API supports the PrometheusAgent CRD", "err ", err)
+		return false, err
+	}
+	if !crdInstalled {
+		level.Info(logger).Log("msg", "Prometheus agent controller disabled because the PrometheusAgent CRD isn't installed")
+		return false, err
+	}
+
+	allowed, err := checkAgentAuthorization(ctx, kclient, conf)
```

Comment:
same here I'd also move `checkAgentAuthorization`  to the pkg/k8sutil package for later re-use. 

````
func CheckPermissions(
  ctx context.Context,
  kclient *kubernetes.Clientset,
  attrs []authv1.ResourceAttributes,  # list of attributes to check
  isAllowedFn func(authv1.ResourceAttributes, bool), # function called for every attribute to tell if it's allowed or not
) error {
  ...
}
````